module/modmanager/manager_test.go

```diff
@@ -678,7 +678,7 @@ func TestModuleReloading(t *testing.T) {
 		test.That(t, logs.FilterMessageSnippet("Module successfully restarted").Len(),
 			test.ShouldEqual, 0)
 		test.That(t, logs.FilterMessageSnippet("Error while restarting crashed module").Len(),
-			test.ShouldEqual, 3)
+			test.ShouldBeGreaterThanOrEqualTo, 1)
```

Comment:
A rare inspiration of reviewing the test for me. I know you didn't write this, but being it was touched, now I'm curious:
* what's the test setup that results in the module restart to fail?
* Is this log line guaranteed to happen before the assertion or is this a race?

Reply:
(hidden diff -- GitHub PRs is not a professional CR product)
```
		// Remove testmodule binary, so process cannot be successfully restarted
		// after crash.
```

(thanks to the later test to inspire me to look for filepath manipulation -- Windows lets one remove a file being executed?). Consider the first question answered.

Still curious about the second question.

Reply:
This isn't a race because the [`WaitForAssertion`][1] earlier in the test is waiting for a line that comes after all the lines in this group of assertions. If we made it past that we're guaranteed that this set of assertions should be accurate.

[1]: https://github.com/viamrobotics/rdk/pull/4997/files#diff-32c15f92eba29518fe6de6907202f8fef5cf52320286f9673df9ad5858371a17R662-R666

Reply:
Ah, I saw that, but now it makes sense. Because* the module won't restart, we go ahead with removing (permanently).